server/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -515,12 +515,23 @@ private static void validateSettings(String name,
         }
     }
 
-    private Settings filterSettings(Settings settings, List<String> settingsFilter) {
+    @VisibleForTesting
+    static Settings filterSettings(Settings settings, List<Setting> settingsFilter) {
         Settings.Builder settingsBuilder = Settings.builder();
-        for (String settingName : settingsFilter) {
-            String setting = settings.get(settingName);
-            if (setting != null) {
-                settingsBuilder.put(settingName, setting);
+        for (Setting setting : settingsFilter) {
+            if (setting.isGroupSetting()) {
+                String prefix = setting.getKey(); // getKey() returns prefix for a group setting
+                var settingsGroup = settings.getByPrefix(prefix);
+                if (settingsGroup.isEmpty() == false) {
+                    for (String name : settingsGroup.keySet()) {
+                        settingsBuilder.put(prefix + name, settingsGroup.get(name)); // No dot added as prefix already has dot at the end.
```

Comment:
minor:
Do we need `if (settingsGroup.isEmpty() == false)` ?
Not sure if we should try to avoid calling `get()`, what do you think?

Reply:
good point, dropped emptiness check.

I tried avoiding `get` by getting a map out of settings via `settingsGroup.getAsStructuredMap()` and then loop over entry set - but then would need to convert Object to String to reflect what `settingsGroup.get(name)` is doing - looked a bit cumbersome.
 
Please note, that this loop is actually one iteration as it's incoming setting specified in `ALTER TABLE SET` and it's a `DDL` operation - I think it should be fine.